rohmu/object_storage/azure.py

```diff
@@ -345,11 +345,14 @@ def store_file_object(self, key, fd, *, cache_control=None, metadata=None, mimet
         content_settings = None
         if mimetype:
             content_settings = ContentSettings(content_type=mimetype)
+        notify_size = [0]
 
         def progress_callback(pipeline_response):
             bytes_sent = pipeline_response.context["upload_stream_current"]
-            if upload_progress_fn and bytes_sent:
-                upload_progress_fn(bytes_sent)
+            if bytes_sent:
+                notify_size[0] = bytes_sent
```

Comment:
This should accumulate the written bytes, not overwrite the previous value.

Reply:
It seems to me that the Azure client already accumulates this internally, though it turns out to be harder to find documentation about this, I wonder how recent the version we're using is. In any case, see line 301 of `/usr/lib/python3.10/site-packages/azure/storage/blob/_shared/policies.py`